packages/datagateway-search/src/table/datasetSearchTable.component.test.tsx

```diff
@@ -1,662 +1,577 @@
+      if (/.*\/facilitycycles$/.test(url)) {
+        return Promise.resolve({
+          data: [
+            {
+              id: 6,
+              name: 'facility cycle name',
+              startDate: '2000-06-10',
+              endDate: '2020-06-11',
+            },
+          ],
+        });
+      }
+      return mockAxiosGet(url);
     });
 
-    const wrapper = createWrapper('isis');
+    renderComponent('isis');
 
-    expect(useDatasetSizes).toHaveBeenCalledWith({ pages: [rowData] });
-    expect(useDatasetsDatafileCount).toHaveBeenCalledWith(undefined);
-
-    expect(wrapper.find('[aria-colindex=3]').find('a').prop('href')).toEqual(
-      `/browse/instrument/4/facilityCycle/6/investigation/2/dataset/1`
-    );
-    expect(wrapper.find('[aria-colindex=3]').text()).toEqual(
-      'Dataset test name'
-    );
-    expect(wrapper.find('[aria-colindex=4]').text()).toEqual('1 B');
+    expect(
+      await screen.findByRole('link', { name: 'Dataset test name' })
+    ).toBeInTheDocument();
+    expect(await screen.findByText('1 B')).toBeInTheDocument();
   });
 
-  it('does not render ISIS link when instrumentId cannot be found', () => {
-    (useAllFacilityCycles as jest.Mock).mockReturnValue({
-      data: [
+  it('does not render ISIS link when instrumentId cannot be found', async () => {
+    const { investigationinstrument, ...data } = mockSearchResults[0].source;
+    searchResponse = {
+      results: [
         {
-          id: 4,
-          name: 'facility cycle name',
-          startDate: '2000-06-10',
-          endDate: '2020-06-11',
+          ...mockSearchResults[0],
+          source: data,
         },
       ],
-    });
-    delete rowData[0].investigation?.investigationInstruments;
+    };
 
-    (useDatasetsInfinite as jest.Mock).mockReturnValue({
-      data: { pages: [rowData] },
-      fetchNextPage: jest.fn(),
-    });
-    const wrapper = createWrapper('isis');
+    renderComponent('isis');
 
-    expect(wrapper.find('[aria-colindex=3]').find('a')).toHaveLength(0);
-    expect(wrapper.find('[aria-colindex=3]').text()).toEqual(
-      'Dataset test name'
-    );
+    await waitFor(async () => {
+      // the title should not be rendered as a link...
+      expect(
+        screen.queryByRole('link', { name: 'Dataset test name' })
+      ).toBeNull();
+      // ...but it should still be rendered as a normal text
+      expect(screen.getByText('Dataset test name')).toBeInTheDocument();
+    });
+    expect(await screen.findByText('1 B')).toBeInTheDocument();
   });
 
-  it('does not render ISIS link when facilityCycleId cannot be found', () => {
-    const wrapper = createWrapper('isis');
-
-    expect(wrapper.find('[aria-colindex=3]').find('a')).toHaveLength(0);
-    expect(wrapper.find('[aria-colindex=3]').text()).toEqual(
-      'Dataset test name'
-    );
-  });
+  it('does not render ISIS link when facilityCycleId cannot be found', async () => {
+    renderComponent('isis');
```

Comment:
Don't we have to remove the `facilityCycleId` from the response first?

Reply:
However, the "wait for query client to finish fetching" technique can be deployed in other tests,  I vaguely remember some tests having the same issue but I can't remember which off the top of my head. I can have a look at it and implement the changes in #1447

Reply:
Fantastic